File: Sdks/ConsumerApi.Sdk/src/Endpoints/SyncRuns/Types/SyncRun.cs
namespace Backbone.ConsumerApi.Sdk.Endpoints.SyncRuns.Types;

public class SyncRun
{
    public required string Id { get; set; }
    public required string Type { get; set; }
    public required DateTime ExpiresAt { get; set; }
    public required long Index { get; set; }
    public required DateTime CreatedAt { get; set; }
    public required string CreatedBy { get; set; }
    public required string CreatedByDevice { get; set; }
    public required int EventCount { get; set; }
}

File: Applications/ConsumerApi/test/ConsumerApi.Tests.Performance/src/libs/backbone-client/models/sync-run.ts
export interface StartSyncRunRequestBody {
    type: SyncRunType;
    duration: number;
}

export enum SyncRunType {
    ExternalEventSync,
    DatawalletVersionUpgrade
}

interface SyncRun {
    id: string;
    expiresAt: string;
    index: number;
    createdAt: string;
    createdBy: string;
    createdByDevice: string;
    eventCount: number;
}

export interface StartSyncRunResponse {
    status: StartSyncRunStatus;
    syncRun: SyncRun | null;
}

declare enum StartSyncRunStatus {
    Created = "Created",
    NoNewEvents = "NoNewEvents"
}
